generate_engagement_analysis_files.py

```diff
@@ -103,3 +106,50 @@
 
         for row in opted_in_uids_per_show.items():
             writer.writerow(row)
+
+    # For each uid generate a sustained engagement map that contains : no of shows participated , their manually
+    # coded demographics and a matrix representation of the shows they have participated in
+    # (1 for show they participated and 0 otherwise.)
+    sustained_engagement_map = {}
+    for uid in engagement_map.keys():
+        basic_str = f" {len(engagement_map[uid]['shows'])}, {engagement_map[uid]['demog']['gender']}," \
+            f" {engagement_map[uid]['demog']['age']}, {engagement_map[uid]['demog']['recently_displaced']}, "
+
+        show_mapping = ""
+        for show_name in rqa_raw_fields:
+            if show_name in engagement_map[uid]["shows"]:
+                show_mapping = show_mapping + "1, "
```

Comment:
Why are you reading and writing show_mapping with string manipulation rather than lists or dicts?

Reply:
This allows us to have a _"matrix"_ representation of uid show participation which is then stored as a list.

Reply:
Discussed offline. Have removed this piece as its no longer relevant

Reply:
I have refactored this to loop through engagement map to make it less complicated.